src/test/java/com/squareup/javapoet/MemberRefTest.java

```diff
@@ -0,0 +1,86 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.squareup.javapoet;
+
+import static com.google.common.truth.Truth.assertThat;
+
+import java.lang.reflect.Method;
+import java.lang.reflect.Type;
+import java.util.concurrent.TimeUnit;
+
+import javax.lang.model.element.Modifier;
+
+import org.junit.Test;
+
+public class MemberRefTest {
+  @Test public void readmeExampleWithThreadStateNew() {
+    CodeBlock block = CodeBlock.builder()
+        .add("$R", Thread.State.NEW)
+        .build();
+    assertThat(block.toString()).isEqualTo("java.lang.Thread.State.NEW");
+  }
+
+  @Test public void readmeExampleWithTimeUnitConvert() throws Exception {
+    Method convert = TimeUnit.class.getMethod("convert", long.class, TimeUnit.class);
+    MethodSpec.Builder method = MethodSpec.methodBuilder("minutesToSeconds")
+        .returns(long.class)
+        .addParameter(long.class, "minutes")
+        .addStatement("return $R(minutes, $R)",
+            MemberRef.get(convert, MemberRef.get(TimeUnit.SECONDS)),
+            TimeUnit.MINUTES);
+    String unit = "java.util.concurrent.TimeUnit";
+    assertThat(method.build().toString()).isEqualTo(""
+        + "long minutesToSeconds(long minutes) {\n"
+        + "  return " + unit + ".SECONDS.convert(minutes, " + unit + ".MINUTES);\n"
+        + "}\n");
+    TypeSpec util = TypeSpec.classBuilder("Util")
+        .addMethod(method
+            .addModifiers(Modifier.PUBLIC, Modifier.STATIC)
+            .build())
+        .build();
+    assertThat(JavaFile.builder("readme", util).build().toString()).isEqualTo(""
+        + "package readme;\n"
+        + "\n"
+        + "import java.util.concurrent.TimeUnit;\n"
+        + "\n"
+        + "class Util {\n"
+        + "  public static long minutesToSeconds(long minutes) {\n"
+        + "    return TimeUnit.SECONDS.convert(minutes, TimeUnit.MINUTES);\n"
+        + "  }\n"
+        + "}\n");
+  }
+
+  @Test public void readmeExampleWithMethodChaining() throws Exception {
+    // get members
+    Method cbBuilder = CodeBlock.class.getMethod("builder"); // static
+    Method cbAdd = CodeBlock.Builder.class.getMethod("add", String.class, Object[].class);
+    Method cbBuild = CodeBlock.Builder.class.getMethod("build");
+    // create refs explicitly
+    TypeName tnCodeBlock = TypeName.get(CodeBlock.class);
+    MemberRef mrBuilder = MemberRef.get(cbBuilder, tnCodeBlock, new Type[0]);
+    MemberRef mrAdd = MemberRef.get(cbAdd, "");
+    MemberRef mrBuild = MemberRef.get(cbBuild, "");
```

Comment:
Overloaded `get` too much. Having to pass an empty array is not ... nice.

Will introduce more specific getters, like `getStaticMemberRef`, `getInstanceMemberRef`